xrpl4j-core/src/main/java/org/xrpl/xrpl4j/crypto/keys/PrivateKey.java

```diff
@@ -32,41 +33,189 @@
+      value.length() == 32,
+      "Byte values passed to this constructor must be 32 bytes long, with no prefix."
+    );
+
+    this.value = UnsignedByteArray.of(value.toByteArray()); // <- Always copy to ensure immutability
   }
 
   /**
    * Accessor for the key value, in binary (Note: will be 33 bytes).
    *
    * @return An instance of {@link UnsignedByteArray}.
+   *
+   * @deprecated Prefer {@link #valueWithPrefixedBytes()} or {@link #valueWithNaturalBytes()} instead.
    */
+  @Deprecated
   public UnsignedByteArray value() {
-    return UnsignedByteArray.of(value.toByteArray());
+    // Check for empty value, which can occur if this PrivateKey is "destroyed" but still in memory.
+    if (value.length() == 0) {
+      return UnsignedByteArray.empty();
+    } else {
+      // This is technically wrong (because `value()` had an ambiguous meaning prior to fixing #486), but this mirrors
+      // what's in v3 prior to fixing #486, and will be fixed in v4 once the deprecated `.value()` is removed.
+      return this.valueWithPrefixedBytes();
+    }
+  }
+
+  /**
+   * Accessor for the byte value in {@link #value()} but in a more natural form (i.e., the size of the returned value
+   * will be 32 bytes). Natural ed25519 or secp256k1 private keys will ordinarily contain only 32 bytes. However, in
+   * XRPL, private keys are represented with a single-byte prefix (i.e., `0xED` for ed25519 and `0x00` for secp256k1
+   * keys).
+   *
+   * @return An instance of {@link UnsignedByteArray}.
+   */
+  public UnsignedByteArray valueWithNaturalBytes() {
+    // Check for empty value, which can occur if this PrivateKey is "destroyed" but still in memory.
+    if (value.length() == 0) {
+      return UnsignedByteArray.empty();
+    } else {
+      // Note: `toByteArray()` will perform a copy, which is what we want in order to enforce immutability of this
+      // PrivateKey (because Java 8 doesn't support immutable byte arrays).
+      return UnsignedByteArray.of(value.toByteArray());
+    }
+  }
+
+  /**
+   * Accessor for the byte value in {@link #value()} but in a more natural form (i.e., the size of the returned value
+   * will be 32 bytes). Natural ed25519 or secp256k1 private keys will ordinarily contain only 32 bytes. However, in
+   * XRPL, private keys are represented with a single-byte prefix (i.e., `0xED` for ed25519 and `0x00` for secp256k1
+   * keys).
+   *
+   * @return An instance of {@link UnsignedByteArray}.
+   */
+  public UnsignedByteArray valueWithPrefixedBytes() {
+    // Check for empty value, which can occur if this PrivateKey is "destroyed" but still in memory.
+    if (value.length() == 0) {
+      return UnsignedByteArray.empty();
+    } else {
+      // Note: value.slice() will take a view of the existing UBA, then `.toByteArray()` will perform a copy, which is
+      // what we want in order to enforce immutability of this PrivateKey (because Java 8 doesn't support immutable byte
+      // arrays).
```

Comment:
I think this comment doesn't apply here anymore. Maybe it should go on line 96?

Reply:
Fixed via 92fb265f95d77220f2c567d242e3de867ad50c0c